mono/metadata/assembly.c

```diff
@@ -557,6 +557,21 @@ mono_assembly_getrootdir (void)
 }
 
 /**
+ * mono_native_getrootdir:
+ * 
+ * Obtains the root directory used for looking up native libs (.so, .dylib).
+ *
+ * Returns: a string with the directory, this string should be freed by
+ * the caller.
+ */
+G_CONST_RETURN gchar *
```

Comment:
This entire patch is @directhex's code originally, so I defer to him. I'm happy to amend this PR, but I can't speak to it's correctness other than to say this is already committed in master.